### ISSUE TITLE
computation of various statistics maybe wrong because `self.len` != `self.total` when fixed capacity container is used

### ISSUE THREAD

**Issue by dilawar** [SubconsciousCompute/SimpleAccumulator#19]
`self.len` is the total number of elements only when underlying container stores all elements. When we use a circular buffer, `self.len` is not total number of elements.

https://github.com/SubconsciousCompute/SimpleAccumulator/blob/f3bc60fffc0fbca80c82e64242524ee3e9dc4bde/src/lib.rs#L248

## Testing with watermill 

- `variance` seems correct.
- `median` is not correct.

cc: @sn99

**Comment by dilawar** [comment]
Looks like most statistics are wrong when fixed capacity accumulator is used. 

```
/// Compare with third-party implementation.
#[test]
fn test_sanity() {
    use watermill::quantile::Quantile;
    use watermill::stats::Univariate;

    let mut acc = SimpleAccumulator::with_fixed_capacity::<f64>(&vec![], 10, true);
    let mut running_median: Quantile<f64> = Quantile::new(0.5f64).unwrap();

    let total = 1000;
    for _i in 0..total {
        let a = 100.0 * (rand::random::<f64>() - 0.5);
        acc.push(a);
        running_median.update(a);
        println!("{a}");
    }
    println!("{acc:#?}");
    assert_eq!(acc.total, total);
    assert_eq!(running_median.get(), acc.median);
}
```

@sn99 We need to unpublish all version that has new statistics.

**Comment by sn99** [comment]
Our median was always a rough estimate for fixed capacity, there was no way to calculate it in constant time otherwise. We mentioned it in readme https://github.com/SubconsciousCompute/SimpleAccumulator.

> We calculate approx median((min + max + 2*mean)/4) as to not iterate again and again but you can set the exact one by using calculate_median().

**Comment by dilawar** [comment]
Right. We are now using quantile as median estimator https://crates.io/crates/watermill has example in README.md

**Comment by sn99** [comment]
> Right. We are now using quantile as median estimator https://crates.io/crates/watermill has example in README.md. Here are values are way too off from each other.

@dilawar it uses sorting internally https://docs.rs/watermill/0.1.1/src/watermill/quantile.rs.html#155-184, I think that is bound to have a performance impact

**Comment by purnatag** [comment]
`self.len` is supposed to be unrealted to `self.total` in `fixed-capacity` since the idea of `self.total` is to keep a track of all the elements seen so far, in order to calculate running mean and variance. And we had discussed long ago that only mean and variance are running stats, rest are measures on the accumulator that aid in determining bimodality. If we need the other measures to also be running stats, we can just use unbounded capacity.

**Comment by dilawar** [comment]
Fixed in 0.6.0.
